src/Auth0.AuthenticationApi/Tokens/IdTokenValidator.cs

```diff
@@ -33,9 +33,8 @@ internal static async Task AssertTokenMeetsRequirements(this IdTokenRequirements
 
             var token = DecodeToken(rawIDToken);
 
-            // For now we want to support HS256 + ClientSecret as we just had a major release.
-            // TODO: In the next major (v4.0) we should remove this condition as well as Auth0ClientOptions.ClientSecret
-            if (token.SignatureAlgorithm != "HS256")
+            // TODO: Make signatureVerifier non-optional and not skipped for 'HS256' in 7.0.0
+            if (token.SignatureAlgorithm != "HS256" || signatureVerifier != null)
```

Comment:
where is `signatureVerifier` set and on which scenarios? and more importantly what if the token is signed with `alg=none`?

Reply:
That will be in the next PR that actually allows the HS256 validation to be used.